### DISCUSSION TITLE
Remember feature not working

### DISCUSSION THREAD

**Discussion by phddoom** [janko/rodauth-rails#315] [Q&A]
Is anyone else having issues with the remember feature?

My app starts a login loop i.e. keeps redirecting to the login page after redirecting to the root page after logging in, if `after_login { remember_login }` is set.

**Comment by janko** [comment]
Is any additional Rodauth code being run on root page, what do you have in your Rodauth app's `route` block and/or controller callbacks? The official demo app also has `after_login { remember_login }` (for the non-2FA case), and it doesn't produce an infinite redirect loop.

**Reply by phddoom** [reply]
Here is the `app/misc/rodauth_app.rb`

```ruby
class RodauthApp < Rodauth::Rails::App
  # primary configuration
  configure RodauthMain

  # secondary configuration
  # configure RodauthAdmin, :admin

  route do |r|
    rodauth.load_memory # autologin remembered users
    rodauth.check_active_session

    r.rodauth # route rodauth requests

    # ==> Authenticating requests
    # Call `rodauth.require_account` for requests that you want to
    # require authentication for. For example:
    #
    # # authenticate /dashboard/* and /account/* requests
    # if r.path.start_with?("/dashboard") || r.path.start_with?("/account")
    #   rodauth.require_account
    # end

    # ==> Secondary configurations
    # r.rodauth(:admin) # route admin rodauth requests
  end
end
```

As far as I can tell the only rodauth code that should be running on the root page is the code that enforces login.

This is done through a combination of a `before_action`  on the `ApplicationController` and a method that ensures rodauth actions do not require authentication.

```ruby
class ApplicationController < ActionController::Base
  before_action :authenticate!, unless: :public?
  protect_from_forgery with: :null_session,
    if: proc { |c| c.request.format =~ %r{application/json} }

  private ######################################################################

  def authenticate!
    rodauth.require_account
  end

  def current_user
    rodauth.rails_account
  end
  helper_method :current_user

  def public?
    false
  end
end
```
```ruby
class RodauthController < ApplicationController
  # Used by Rodauth for rendering views, CSRF protection, running any
  # registered action callbacks and rescue handlers, instrumentation etc.

  # Controller callbacks and rescue handlers will run around Rodauth endpoints.
  # before_action :verify_captcha, only: :login, if: -> { request.post? }
  # rescue_from("SomeError") { |exception| ... }

  # Layout can be changed for all Rodauth pages or only certain pages.
  # layout "authentication"
  # layout -> do
  #   case rodauth.current_route
  #   when :login, :create_account, :verify_account, :verify_account_resend,
  #        :reset_password, :reset_password_request
  #     "authentication"
  #   else
  #     "application"
  #   end
  # end
  private ######################################################################

  def public?
    true
  end
end

```

**Reply by janko** [reply]
I created a fresh Rails app, installed Rodauth, added `active_sessions` and copy-pasted only the code that you posted, and I don't get the infinite redirect loop on login. Any chance you could reproduce the issue in a fresh Rails app?

**Reply by phddoom** [reply]
I am happy to try, which rails version was your fresh rails app?

**Reply by janko** [reply]
The latest - Rails 7.2.1

**Reply by phddoom** [reply]
I haven't recreated the issue in a fresh app. But I did upgrade the app experiencing the issue from 7.1 to 7.2.1 and that fixed it. Not sure if that means it is a bug in Rails or a bug in Rodauth Rails. Thoughts?

**Reply by janko** [reply]
rodauth-rails should be compatible with the last minor version of each major version starting from Rails 5.0+. I tried downgrading my reproduction app to Rails 7.1, and it still works.
